### PR TITLE
Docs/fixes

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

seems that #5874 introduced some failers on Read-The-Docs
see: https://readthedocs.org/projects/pytorch-lightning/builds/

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/.build_docs.sh [hunk 1]
```diff
@@ -1,3 +0,0 @@
-rm -rf source/generated
```
Comment (justusschock) [inline docs/.build_docs.sh:1]: why delete this file?

File: pytorch_lightning/utilities/imports.py [hunk 2]
```diff
@@ -39,19 +40,21 @@ def _module_available(module_path: str) -> bool:
         return False
 
 
-def _get_version(package: str) -> LooseVersion:
-    return LooseVersion(pkg_resources.get_distribution(package).version)
+def _compare_version(package: str, op, version) -> bool:
```
Comment (justusschock) [inline pytorch_lightning/utilities/imports.py:43]:
```suggestion
def _compare_version(package: str, op: Callable, version: str) -> bool:
```

Comment (justusschock) [reply pytorch_lightning/utilities/imports.py:43]: ofc you have to import callable then :)

**Review by justusschock — APPROVED** [review]

File: docs/.build_docs.sh (hunk 1 above)
Comment (Borda) [reply docs/.build_docs.sh:1]: you were using it?
it was duplicate to `make docs` in root

File: pytorch_lightning/utilities/imports.py [hunk 3]
```diff
@@ -12,12 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """General utilities"""
+import operator
 import platform
 from distutils.version import LooseVersion
 from importlib.util import find_spec
```
Comment (Borda) [inline pytorch_lightning/utilities/imports.py:18]:
```suggestion
from importlib.util import find_spec
from typing import Callable
```

**Review by tchaton — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/utilities/imports.py [hunk 4]
```diff
@@ -39,19 +40,21 @@ def _module_available(module_path: str) -> bool:
...
+    try:
+        pkg_version = LooseVersion(get_distribution(package).version)
+        return op(pkg_version, LooseVersion(version))
+    except DistributionNotFound:
+        return False
 
 
 _IS_WINDOWS = platform.system() == "Windows"
-_TORCH_GREATER_EQUAL_1_6_0 = _get_version("torch") >= LooseVersion("1.6.0")
+_TORCH_GREATER_EQUAL_1_6 = _compare_version("torch", operator.ge, "1.6.0")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:52]: This should be 1_6_0 for consistency with conditions for micro releases

**Review by carmocca — COMMENTED** [review]
Note that the docs failure doesnt have anything to do with #5874

**Pull request merged**